packages/clerk-js/src/ui/components/OrganizationProfile/MembersSearchForm.tsx

```diff
@@ -1,89 +1,84 @@
-   */
-  value: string;
   /**
    * Paginated organization memberships
    */
   memberships: ReturnType<typeof useOrganization>['memberships'];
   /**
-   * Handler for change event on input field
+   * Handler for value changes
    */
-  onSearchChange: (value: string) => void;
+  onChange: (query: string) => void;
   /**
-   * Handler for `query` value changes
+   * Minimum search length to trigger query
    */
-  onQueryTrigger: (query: string) => void;
+  minLength: number;
 };
 
-const membersSearchDebounceMs = 500;
-
-export const MembersSearch = ({ query, value, memberships, onSearchChange, onQueryTrigger }: MembersSearchProps) => {
+export const MembersSearchForm = ({ memberships, onChange, minLength }: MembersSearchProps) => {
   const { t } = useLocalizations();
-
-  const debounceTimer = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const searchField = useFormControl('search', '', {
+    type: 'search',
+    label: '',
+    placeholder: localizationKeys('organizationProfile.membersPage.action__search'),
+  });
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const eventValue = event.target.value;
-    onSearchChange(eventValue);
+    const value = event.target.value.trim();
 
-    const shouldClearQuery = eventValue === '';
-    if (shouldClearQuery) {
-      onQueryTrigger(eventValue);
-    }
-  };
+    searchField.onChange(event);
+    onChange(value);
 
-  // Debounce the input value changes until the user stops typing
-  // to trigger the `query` param setter
-  function handleKeyUp() {
-    if (debounceTimer.current) {
-      clearTimeout(debounceTimer.current);
+    if (value.length >= minLength) {
+      searchField.clearFeedback();
+      return;
     }
 
-    debounceTimer.current = setTimeout(() => {
-      onQueryTrigger(value.trim());
-    }, membersSearchDebounceMs);
-  }
+    searchField.setInfo(t(localizationKeys('organizationProfile.membersPage.action__search_minLength')));
+  };
 
   // If search is not performed on a initial page, resets pagination offset
   // based on the response count
   useEffect(() => {
-    if (!query || !memberships?.data) {
+    if (!searchField.value || !memberships?.data) {
       return;
     }
 
     const hasOnePageLeft = (memberships?.count ?? 0) <= ACTIVE_MEMBERS_PAGE_SIZE;
     if (hasOnePageLeft) {
       memberships?.fetchPage?.(1);
     }
-  }, [query, memberships]);
+  }, [searchField.value, memberships]);
 
-  const isFetchingNewData = value && !!memberships?.isLoading && !!memberships.data?.length;
+  const isFetchingNewData = searchField.value && !!memberships?.isLoading && !!memberships.data?.length;
 
   return (
-    <Animated asChild>
-      <Flex sx={{ width: '100%' }}>
+    <Flex
+      sx={{ minWidth: '50%', position: 'relative' }}
+      direction='col'
+    >
+      <Field.Root {...searchField}>
         <InputWithIcon
-          value={value}
+          {...searchField.props}
```

Comment:
This is resulting in a bunch of warnings for passing invalid dom props.

![Screenshot 2025-02-14 at 11 26 57 AM](https://github.com/user-attachments/assets/6350c007-6d90-4c85-a705-51c53550eb48)

Reply:
Pushed a fix on https://github.com/clerk/javascript/pull/5159/commits/4a53af33e705f16b9ef4815b0efbf0afa19bd2cf - we currently only rely on the `blur` + `focus` event handlers from the `useFormControl` context